### ISSUE TITLE
Rule S3928: FP when using record parameters

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#5094]
### Description

[Rule S3928](https://rules.sonarsource.com/csharp/RSPEC-3928) is raising FPs when using record parameters.

### Repro steps

```
using System;

public record Data(string Code)
{
    public string Code { get; } = Code ?? throw new ArgumentNullException(nameof(Code)); // Noncompliant, FP - Code is a record parameter
}
```

### Expected behavior

For the given example there should be no issue raised since the `Code` parameter is defined in the `Data` record parameter list.

### Actual behavior

An issue is raised: "The parameter named `Code` is not declared in the argument list.".

### Known workarounds

Refactor the code to avoid using of Record parameters.

### Related information

* C#/VB.NET Plugins version: [8.32](https://github.com/SonarSource/sonar-dotnet/releases/tag/8.32.0.39516)

**Comment by simonvane** [comment]
Any idea on timescales for fixing this?
We are using this technique to avoid nulls in deserialized records more and more.
Thanks.

**Comment by andrei-epure-sonarsource** [comment]
@simonvane this fix has been included in the latest release

**Comment by simonvane** [comment]
That's great. Thanks a lot.

**Comment by simonvane** [comment]
Hi @andrei-epure-sonarsource,
Has this been released yet? We are still getting warnings in SonarCloud and SonarLint.
Thanks.

**Comment by pavel-mikula-sonarsource** [comment]
Hi @simonvane,

8.44 is not on SonarCloud yet.
You can track this issue to know when it will be embedded into the next update of SonarLint.
These products have independent release cycles.

You might have found a new case. Please create a new issue for it once you confirm it on 8.44 version too.

**Comment by costin-zaharia-sonarsource** [comment]
@simonvane and @pavel-mikula-sonarsource: I manually tested the case and the issue is not reproducible on master. I suggest waiting for `8.44` to be deployed on Sonar Cloud and [embedded in SonarLint](https://github.com/SonarSource/sonarlint-visualstudio/issues/3196).

**Comment by andrei-epure-sonarsource** [comment]
@simonvane you can verify the version deployed on SonarCloud here: https://sonarcloud.io/api/plugins/installed
Search for "sonar-csharp-plugin-"

**Comment by marisks** [comment]
I see that it is updated in VS SonarLint plugin but not in Rider (IntelliJ) SonarLint. When could it be done?

**Comment by andrei-epure-sonarsource** [comment]
this is the repository for the Rider plugin: https://github.com/SonarSource/sonarlint-omnisharp/releases

I'll ask the team when they plan their next release
